### ISSUE TITLE
Invalid argument upon commiting image on CentOS 7.4

### ISSUE THREAD

**Issue by FireDrunk** [projectatomic/buildah#655]
**Description**
Error when trying to commit a container.
**Steps to reproduce the issue:**
My buildscript:
```
#!/bin/bash -ex

# Settings
NAME=buildah-arch

# Prereqs
ORGA=archimg
TAR_DATE=${TAR_DATE:-$(date +'%Y.%m.01')}
ARCHLINUX_ARCH=x86_64
ARCHLINUX_MIRR=https://archive.archlinux.org/iso

mkdir -p base/arch-rootfs
mkdir -p tarballs/${TAR_DATE}

(
cd tarballs/${TAR_DATE} \
&& wget -q --no-cookies \
	${ARCHLINUX_MIRR}/${TAR_DATE}/archlinux-bootstrap-${TAR_DATE}-${ARCHLINUX_ARCH}.tar.gz{,.sig} \
	${ARCHLINUX_MIRR}/${TAR_DATE}/sha1sums.txt \
&& grep archlinux-bootstrap-${TAR_DATE}-${ARCHLINUX_ARCH}.tar.gz sha1sums.txt \
	| sha1sum --check -
)

tar xz \
	-f tarballs/${TAR_DATE}/archlinux-bootstrap-${TAR_DATE}-${ARCHLINUX_ARCH}.tar.gz \
	--strip-components=1 \
	--directory=./base/arch-rootfs \
        root.${ARCHLINUX_ARCH}/

# Buildah
buildah delete ${NAME} || echo "No image found"

buildah from --name ${NAME} scratch
ROOT=$(buildah mount ${NAME})

buildah copy ${NAME} base/arch-rootfs/ /

buildah run ${NAME} chmod 755 /etc
buildah run ${NAME} pacman-key --init 
buildah run ${NAME} pacman-key --populate archlinux
sed -i 's/#Server = http:\/\/mirror.ams1.nl.leaseweb.net\/archlinux\/$repo\/os\/$arch/Server = http:\/\/mirror.ams1.nl.leaseweb.net\/archlinux\/$repo\/os\/$arch/g' ${ROOT}/etc/pacman.d/mirrorlist
buildah run ${NAME} pacman --noconfirm -Syu --needed base
buildah run ${NAME} pacman --noconfirm -Rsc cryptsetup device-mapper dhcpcd iproute2 jfsutils linux lvm2 man-db man-pages mdadm nano netctl openresolv pciutils pcmciautils reiserfsprogs s-nail systemd-sysvcompat usbutils vi xfsprogs 
buildah run ${NAME} rm -rf /usr/share/man/* /var/cache/pacman/pkg/* /var/lib/pacman/sync/* /README /etc/pacman.d/mirrorlist.pacnew
buildah config --cmd "/bin/bash" ${NAME} 
```
This runs fine, I know it's not the most perfect script, it's a friday-afternoon-hackathon thingy :-)
Afterwards, when running: "buildah commit buildah-arch buildah-arch-done" I get an error:

**Describe the results you received:**
```
[root@mgt buildah-arch]# buildah commit buildah-arch buildah-arch
error committing container "buildah-arch" to "buildah-arch": error copying layers and metadata: Error initializing source containers-storage:buildah-arch: error extracting layer "1e9dec5e5d4a857b15b1d0e292d3cfd284f713225e9948823a4d7111f71654b0": devmapper: Error mounting '/dev/mapper/container-253:0-33636291-1e9dec5e5d4a857b15b1d0e292d3cfd284f713225e9948823a4d7111f71654b0' on '/var/lib/containers/storage/devicemapper/mnt/1e9dec5e5d4a857b15b1d0e292d3cfd284f713225e9948823a4d7111f71654b0': invalid argument
<7>[14081.560303] device-mapper: ioctl: unable to remove open device container-253:0-33636291-pool
<4>[14090.799257] SELinux: mount invalid.  Same superblock, different security settings for (dev dm-3, type xfs)

[root@mgt buildah-arch]# buildah commit buildah-arch buildah-arch-done
error committing container "buildah-arch" to "buildah-arch-done": error copying layers and metadata: Error initializing source containers-storage:buildah-arch: error extracting layer "1e9dec5e5d4a857b15b1d0e292d3cfd284f713225e9948823a4d7111f71654b0": devmapper: Error mounting '/dev/mapper/container-253:0-33636291-1e9dec5e5d4a857b15b1d0e292d3cfd284f713225e9948823a4d7111f71654b0' on '/var/lib/containers/storage/devicemapper/mnt/1e9dec5e5d4a857b15b1d0e292d3cfd284f713225e9948823a4d7111f71654b0': invalid argument
<7>[14090.804216] device-mapper: ioctl: unable to remove open device container-253:0-33636291-pool
<4>[14095.600104] SELinux: mount invalid.  Same superblock, different security settings for (dev dm-3, type xfs)
```

**Describe the results you expected:**
A committed image.

**Output of `rpm -q buildah` or `apt list buildah`:**
I've compiled Buildah from source.

**Output of `buildah version`:**
```
[root@mgt usr]# buildah --version
buildah version 0.16 (image-spec 1.0.0, runtime-spec 1.0.0)
```

**Output of `cat /etc/*release`:**
```
[root@mgt usr]# cat /etc/redhat-release 
CentOS Linux release 7.4.1708 (Core)
```

**Output of `uname -a`:**

```
[root@mgt usr]# uname -a
Linux mgt.cravu.lan 3.10.0-693.21.1.el7.x86_64 #1 SMP Wed Mar 7 19:03:37 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

```

**Output of `cat /etc/containers/storage.conf`:**

```
[root@mgt usr]# cat /etc/containers/storage.conf 
# storage.conf is the configuration file for all tools
# that share the containers/storage libraries
# See man 5 containers-storage.conf for more information

# The "container storage" table contains all of the server options.
[storage]

# Default Storage Driver
driver = "devicemapper"

# Temporary storage location
runroot = "/var/run/containers/storage"

# Primary Read/Write location of container storage
graphroot = "/var/lib/containers/storage"

[storage.options]
# AdditionalImageStores is used to pass paths to additional Read/Only image stores
# Must be comma separated list.
additionalimagestores = [
]

# Size is used to set a maximum size of the container image.  Only supported by
# certain container storage drivers.
size = ""

# OverrideKernelCheck tells the driver to ignore kernel checks based on kernel version
#override_kernel_check = "true"

```

**Comment by FireDrunk** [comment]
I don't have anything else mounted (from buildah):
```
[root@mgt ~]# mount
sysfs on /sys type sysfs (rw,nosuid,nodev,noexec,relatime,seclabel)
proc on /proc type proc (rw,nosuid,nodev,noexec,relatime)
devtmpfs on /dev type devtmpfs (rw,nosuid,seclabel,size=1929640k,nr_inodes=482410,mode=755)
securityfs on /sys/kernel/security type securityfs (rw,nosuid,nodev,noexec,relatime)
tmpfs on /dev/shm type tmpfs (rw,nosuid,nodev,seclabel)
devpts on /dev/pts type devpts (rw,nosuid,noexec,relatime,seclabel,gid=5,mode=620,ptmxmode=000)
tmpfs on /run type tmpfs (rw,nosuid,nodev,seclabel,mode=755)
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,seclabel,mode=755)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,release_agent=/usr/lib/systemd/systemd-cgroups-agent,name=systemd)
pstore on /sys/fs/pstore type pstore (rw,nosuid,nodev,noexec,relatime)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_prio,net_cls)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpuacct,cpu)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
cgroup on /sys/fs/cgroup/hugetlb type cgroup (rw,nosuid,nodev,noexec,relatime,hugetlb)
configfs on /sys/kernel/config type configfs (rw,relatime)
/dev/mapper/centos_template-root on / type xfs (rw,relatime,seclabel,attr2,inode64,noquota)
selinuxfs on /sys/fs/selinux type selinuxfs (rw,relatime)
systemd-1 on /proc/sys/fs/binfmt_misc type autofs (rw,relatime,fd=25,pgrp=1,timeout=0,minproto=5,maxproto=5,direct,pipe_ino=8063)
debugfs on /sys/kernel/debug type debugfs (rw,relatime)
hugetlbfs on /dev/hugepages type hugetlbfs (rw,relatime,seclabel)
mqueue on /dev/mqueue type mqueue (rw,relatime,seclabel)
/dev/sda1 on /boot type xfs (rw,relatime,seclabel,attr2,inode64,noquota)
tmpfs on /run/user/0 type tmpfs (rw,nosuid,nodev,relatime,seclabel,size=388104k,mode=700)
[root@mgt ~]# buildah mount
[root@mgt ~]# 
```
I've changed the storage engine from overlay to devicemapper, perhaps that has something to do with it?

After logging in and out, and trying the script again, it works fine. my buildah commit buildah-arch buildah-arch worked fine now.

```
[root@mgt buildah-arch]# buildah commit buildah-arch buildah-arch
Getting image source signatures
Copying blob sha256:ce23df4c8c435d9d64a58dcb93c4ffa1c711404584a5a92eaab54b549a165501
 285.90 MiB / 285.90 MiB [==================================================] 8s
Copying config sha256:e377f4f3ec4e4aedff22f8924fa7c3be644fb9849b658cc8de3f70cef4a3abe9
 299 B / 299 B [============================================================] 0s
Writing manifest to image destination
Storing signatures
e377f4f3ec4e4aedff22f8924fa7c3be644fb9849b658cc8de3f70cef4a3abe9
```
Perhaps some stale reference was still there?

**Comment by rhatdan** [comment]
I would stick to overlay.  The error is curious, since we thought we fixed it.   
The problem is if you mount an file system like devicemapper or overly using an SELinux label like
system_u:object_r:container_file_t:s0:c1,c2 and then attempt during commit to mount the same file system a second time, but this time you use a different label, or no label the kernel will block the mount, whether SELinux is in enforcing mode or permissive. The kernel does not want the data available via two different labels at the same time.  We have this issue with Docker, which I am trying to fix, but we already fixed it in containers/storage (buildah, podman, CRI-O).  Or at least I think we did.
